network-contracts/contracts/MockERC20.sol

```diff
@@ -0,0 +1,30 @@
+pragma solidity >=0.5.0 <0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "./IERC20.sol";
+
+contract MockERC20 is IERC20 {
+    function totalSupply() public view returns (uint256) {
+        return 1000;
+    }
+
+    function balanceOf(address /*account*/) public view returns (uint256) {
+        return 1000;
+    }
+
+    function transfer(address /*recipient*/, uint256 amount) public returns (bool) {
+        return (amount <= 1000);
+    }
+
+    function allowance(address /*owner*/, address /*spender*/) public view returns (uint256) {
+        return 0;
+    }
+
+    function approve(address /*spender*/, uint256 /*amount*/) public returns (bool) {
+        return false;
+    }
+
+    function transferFrom(address /*sender*/, address /*recipient*/, uint256 /*amount*/) public returns (bool) {
+        return false;
+    }
+}
```

Comment:
I like the idea of having a mock but in this case we would need a different contract for each test we will add.
maybe we can have a real one like we do here? https://github.com/status-im/gift-keycard/tree/master/contracts
and then we just setup the state of the contract from the js test

Reply:
if you want to change it, it is ok for me. I don't think this is a priority until the need actually arises though, and the mock is very simple

Reply:
ok!